frame/contracts/src/lib.rs

```diff
@@ -930,6 +932,27 @@ struct InstantiateInput<T: Config> {
 	salt: Vec<u8>,
 }
 
+pub enum CollectEvents {
+	/// # Note
+	///
+	/// Events should only be collected when called off-chain, as this would
+	/// collect all the Events emitted in the block so far and put them in them into the PoV.
+	///
+	/// **Never** use this mode for on-chain execution.
+	UnsafeCollect,
+	/// Events should not be collected.
+	Skip,
```

Comment:
```suggestion
    /// Collect events.
    ///
	/// # Note
	///
	/// Events should only be collected when called off-chain, as this would
	/// collect all the Events emitted in the block so far and put them in them into the PoV.
	///
	/// **Never** use this mode for on-chain execution.
	UnsafeCollect,
	/// Skip event collection.
	Skip,
```